### PR TITLE
Port Filtering Plugin

### PR DESCRIPTION
**Pull request by PSNAppz**
Enabling port-based request filtering in ShadowGuard. 

Key features include:

- Port Blacklist: Block access to specific ports.
- Port Whitelist: Allow access only to specific ports.
- Active Mode: Enforce filtering rules dynamically.
- Database Integration: Log blocked requests for auditing.
- Publisher Notifications: Alert configured systems on blocked requests.
- Configuration is supported via port-blacklist and port-whitelist settings in config.json. Unit tests validate blacklist, whitelist, and default behaviors. 

This plugin enhances ShadowGuard’s security by providing granular control over port-based access.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
 	"database": {
 		"host": "localhost",
 		"port": "5432",
-		"user": "gorm",
-		"password": "gorm",
-		"dbname": "gorm"
+		"user": "postgres",
+		"password": "postgres",
+		"dbname": "shadowguard"
```
Comment (KingAkeem) [inline config.json:7]: You must update the build scripts to reflect these changes so the database is built correctly.

**Review by KingAkeem — CHANGES_REQUESTED** [review]
The build script will break if the config.json is changed currently.

File: config.json (hunk 1 above)
Comment (PSNAppz) [reply config.json]: Fixed!

Comment (KingAkeem) [reply config.json]: Perfect, looks good to me

**Review by KingAkeem — APPROVED** [review]

**Pull request merged**